### DISCUSSION TITLE
Why Tooltip not working with Remix?

### DISCUSSION THREAD

**Discussion by ndozhh** [adobe/react-spectrum#6668] [Q&A] [answered]
Hi everyone,

I'm using RAC Tooltip in a remix project and the simply doesn't work. I have only the following code in my `root.tsx` and it doesn't add the `data-overlay-container` to the DOM when hover or focus the button so the tooltip never shows as you can see it in the first video. I tried the same in a fresh project in Stackblitz and it works well (second video), so I don't know what else to debug in my app. I'm using the latest version of RAC and Remix and also other components are working good, the problem is just with the Tooltip.

Any ideas or suggestions? Thank you!!

```jsx
import { Button, Tooltip, TooltipTrigger } from 'react-aria-components'

<html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body>
        <TooltipTrigger closeDelay={0} delay={0}>
          <Button>Hover me</Button>
          <Tooltip>Random text</Tooltip>
        </TooltipTrigger>
        <Scripts />
      </body>
    </html>
```

Not working
[5ed5160d-b2dd-440d-a0cb-d1bd12428427.webm](https://github.com/adobe/react-spectrum/assets/25773289/16260dbe-9070-42fa-958b-8ff185b20d9c)

Working in Stackblitz
[31c275f6-a056-4f15-9264-947f83637437.webm](https://github.com/adobe/react-spectrum/assets/25773289/71f3f9e1-c5eb-48fc-8a86-bfbd742528cd)

**Comment by snowystinger** [comment]
You should be able to download the stackblitz as a new project. Then I'd see if that project works locally. If it does, then I'd start comparing the settings in your project and the fresh one you downloaded from stackblitz and tweak the new one until it stops working or you can swap it into your project.

I haven't used Remix very much myself and I'm not aware of any issues with it right now.

You might check any caches as well.

**Reply by ndozhh** [reply]
Hello @snowystinger 

I'm trying to make [this example work](https://github.com/adobe/react-spectrum/issues/5733#issuecomment-1918691983) in my remix app but it doesn't. I tested it on [this codesanbox](https://codesandbox.io/p/sandbox/lucid-meninsky-w5qxjc?file=%2Fsrc%2FApp.js%3A27%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clw9g1l3300073581v83b72mm%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clw9g1l3300033581oq870vx2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clw9g1l3300043581cv2osxla%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clw9g1l330006358194f4cuvh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B51.95171282612209%252C48.04828717387791%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clw9g1l3300033581oq870vx2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clw9g1l3300023581wjb8ckxj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A27%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A27%252C%2522endColumn%2522%253A1%257D%255D%257D%252C%257B%2522id%2522%253A%2522cly8v3wm80002358dh180g5tq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A8%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A8%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clw9g1l3300033581oq870vx2%2522%252C%2522activeTabId%2522%253A%2522clw9g1l3300023581wjb8ckxj%2522%257D%252C%2522clw9g1l330006358194f4cuvh%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clw9g1l3300053581eck21o6y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clw9g1l330006358194f4cuvh%2522%252C%2522activeTabId%2522%253A%2522clw9g1l3300053581eck21o6y%2522%257D%252C%2522clw9g1l3300043581cv2osxla%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clw9g1l3300043581cv2osxla%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A27.663085188027623%257D) and it's working but I noticed that the `buttonProps` from `useButton` in my case has less properties that the one in codesandbox:

Mine
![image](https://github.com/adobe/react-spectrum/assets/25773289/544fc1ed-8e76-45ab-b1e3-e4cd611d7eb9)

Codesandbox
![image](https://github.com/adobe/react-spectrum/assets/25773289/5c607b57-1c7c-444e-bf13-e307e4a00641)

These are my imports
```js
import { mergeProps, useButton } from 'react-aria'
import * as RAC from 'react-aria-components'
```

Versions
```json
"react-aria": "3.33.1",
"react-aria-components": "1.2.1",
```
Any idea why? Maybe something special that react-aria does under the hood that is conflicting with Remix for the tooltip?, although I'm usign other components like Button and it works well

**Reply by snowystinger** [reply, marked as answer]
You could check for duplicates/multiple copies of our packages being installed. This usually manifests as a node modules directory inside of another node modules directory. It commonly causes issues like you're seeing.

I'm on my phone but there's a script you can run to help determine this. I put it in an issue recently if you want to have a search for it.

**Reply by ndozhh** [reply]
@snowystinger Man thank you very much!! I found and ran  [the script](https://gist.github.com/jluyau/9024db3527788030312332075745469b) and indeed my monorepo had different versions of some `@react-aria` packages, after update those my initial Tooltip implementation is working now 🚀
